### PR TITLE
ensure things are properly signed before uploading to stage

### PR DESCRIPTION
**Pull request by evgeni**

### CODE REVIEW

File: upload_stage_rpms [hunk 1]
```diff
@@ -5,4 +5,6 @@
 USER='yumrepostage'
 HOST='web01.osuosl.theforeman.org'
 
+./verify_stage_sigs
```
Comment (evgeni) [inline upload_stage_rpms:8]: this will break nightly, huh?

File: verify_stage_sigs [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+set -e
+
+. settings
+
+ARCHES="x86_64 source"
+
+EXIT_CODE=0
+
+for os in $OSES; do
+  BASE="tmp/$PROJECT/$VERSION/$os"
```
Comment (ekohl) [inline verify_stage_sigs:12]: Can we make this a parameter? I don't like that we're spreading `tmp/$PROJECT/$VERSION` all over the code base as hardcoded paths. Looks like `upload_stage_rpms` already has a path `tmp/$PROJECT/$VERSION/` so can we make that `$1`?

Or a new setting?

File: verify_stage_sigs [hunk 3]
```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+
+set -e
+
+. settings
+
+ARCHES="x86_64 source"
```
Comment (ekohl) [inline verify_stage_sigs:7]: Should we move this into `settings` as a new global variable that `sign_stage_rpms` then also uses?

**Review by ekohl — COMMENTED** [review]
I debated adding a noop to `sign_stage_rpms` but I think it's not really in the style of the whole repo so I think your approach is fine.

File: verify_stage_sigs (hunk 3 above)
Comment (evgeni) [reply verify_stage_sigs:7]: probably. `build_stage_repository` also uses arches, but doesn't load settings (and uses them slightly differently too)

File: verify_stage_sigs (hunk 2 above)
Comment (evgeni) [reply verify_stage_sigs]: new setting it is!

File: settings [hunk 4]
```diff
@@ -67,6 +68,7 @@
 RPM_PACKAGES=()
 PACKAGING_PR=${PACKAGING_PR:-true}
 GPG_EXPIRE="1y"
+STAGE_LOCAL_BASE="tmp/$PROJECT/$VERSION"
```
[the remark this answers is not in the document]
Comment (evgeni) [reply settings:71]: we could also use `RPMDIR` (which is currently unused) here… questions and choices.

@ehelms do you recall why `tmp/` was used vs the previously existing path?

[the remark this answers is not in the document]
Comment (ekohl) [reply settings:71]: I suspect it was previously used by the other script and @ehelms didn't want to conflict with it, but now that koji is cleaned up it's an unused variable. When I introduced it, it made sense to me to store it in the release itself.

[the remark this answers is not in the document]
Comment (ehelms) [reply settings:71]: > @ehelms do you recall why `tmp/` was used vs the previously existing path?

What previously existing path? I was the first to introduce the notion of generating staging repositories locally :)

[the remark this answers is not in the document]
Comment (evgeni) [reply settings:71]: But not the first one who had to download rpms, sign them and upload the signatures, which we did in RPMDIR

[the remark this answers is not in the document]
Comment (ehelms) [reply settings:71]: Fair. My motivation was that I wanted the generation of the repositories to be co-located, easy to clean up and to mimic the structure they would have on the staging repository server to allow easier testing and verification of them.

[the remark this answers is not in the document]
Comment (evgeni) [reply settings:71]: Aye. It's probably fine being in `tmp` and thrown away later. Probably bit if cleanup we can perform later.

**Review by ekohl — APPROVED** [review]

**Pull request merged**